### PR TITLE
raftstore: add metrics APPLY_TASK_WAIT_TIME_HISTOGRAM.

### LINKED ISSUE #2492: raftstore: add a metric for apply task waiting time
Like REQUEST_WAIT_TIME_HISTOGRAM to check the raftstore channel wait time, we can also add a APPLY_TASK_WAIT_TIME_HISTOGRAM to see the time when we send the apply task and run it in the apply worker.

### PR DESCRIPTION
**Pull request by hicqu**
close #2492 .

### CODE REVIEW

File: src/raftstore/store/worker/apply.rs [hunk 1]
```diff
@@ -1353,15 +1356,15 @@ pub struct Destroy {
...
+    Applies((Vec<Apply>, Instant)),
     Registration(Registration),
     Proposals(Vec<RegionProposal>),
     Destroy(Destroy),
 }
 
 impl Task {
     pub fn applies(applies: Vec<Apply>) -> Task {
-        Task::Applies(applies)
+        Task::Applies((applies, Instant::now()))
```
Comment (siddontang) [inline src/raftstore/store/worker/apply.rs]: use coarse  now

File: src/raftstore/store/worker/apply.rs [hunk 2]
```diff
@@ -1586,7 +1591,11 @@ impl Runner {
 impl Runnable<Task> for Runner {
     fn run(&mut self, task: Task) {
         match task {
-            Task::Applies(a) => self.handle_applies(a),
+            Task::Applies((a, start_time)) => {
+                let elapsed = duration_to_sec(start_time.elapsed()) as f64;
+                self.apply_task_wait_time.observe(elapsed);
```
Comment (siddontang) [inline src/raftstore/store/worker/apply.rs]: use APPLY_TASK_WAIT_TIME_HISTOGRAM directly

File: src/raftstore/store/worker/apply.rs [hunk 3]
```diff
@@ -1353,15 +1353,15 @@ pub struct Destroy {
 
 /// region related task.
 pub enum Task {
-    Applies(Vec<Apply>),
+    Applies((Vec<Apply>, Instant)),
```
Comment (siddontang) [inline src/raftstore/store/worker/apply.rs]: I prefer using Struct here

**Comment by siddontang** [comment]
Rest LGTM

PTAL @BusyJay

**Comment by hicqu** [comment]
@BusyJay PTAL thanks.

**Comment by siddontang** [comment]
Ping @BusyJay

**Comment by BusyJay** [comment]
LGTM

**Comment by siddontang** [comment]
@hicqu 

you should add this to the Grafana in the raft ready row later.

**Comment by hicqu** [comment]
Got it.

**Pull request merged**